spec/02-integration/05-proxy/14-upstream-status-header_spec.lua

```diff
@@ -0,0 +1,142 @@
+local helpers = require "spec.helpers"
+local constants = require "kong.constants"
+
+describe(constants.HEADERS.UPSTREAM_STATUS .. " header", function()
+  local client
+
+  setup(function()
+    local bp = helpers.get_db_utils()
+    assert(helpers.dao:run_migrations())
+
+    local service = bp.services:insert {
+      host = helpers.mock_upstream_host,
+      port = helpers.mock_upstream_port,
+      protocol = helpers.mock_upstream_protocol,
+    }
+    assert(service)
+
+    local route1 = bp.routes:insert {
+      protocols = { "http" },
+      service = service,
+      paths = { "/foo" },
+    }
+    assert(route1)
+
+    local route2 = bp.routes:insert {
+      protocols = { "http" },
+      service = service,
+      paths = { "/bar" },
+    }
+    assert(route2)
+
+    bp.plugins:insert({
+      name = "dummy",
+      route_id = route2.id,
+      config = {
+        resp_code = 500,
+      }
+    })
+  end)
+
+
+  describe("should be same as upstream status code", function()
+    setup(function()
+      assert(helpers.start_kong {
+        headers = "server_tokens,latency_tokens,x-kong-upstream-status",
+        custom_plugins = "dummy",
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+      })
+      client = helpers.proxy_client()
+    end)
+
+    teardown(function()
+      if client then
+        client:close()
+      end
+      helpers.stop_kong(nil, nil, true)
```

Comment:
Using `true` in arguments to `helpers.stop_kong` should be for debugging purposes only. This is a leftover probably? There are other places in this file.

Reply:
This was used to avoid truncating tables between tests runs in this file since `stop_kong` will truncate them by default. Should it still be avoided?